deploy/docker/Dockerfile

```diff
@@ -1,28 +1,44 @@
 FROM alpine:3.5
 
-LABEL maintainer="adundovi@gmail.com"
-LABEL description="Prepares a docker image \
-of pump.io server, http://pump.io/."
+LABEL maintainer Andrej Dundovic <adundovi@gmail.com>
+LABEL description "Stream server that does \
+most of what people really want from a social network."
 
 ENV LC_ALL en_US.UTF-8
 
+ENV PUMPIO_VERSION 3.0.2
+
 RUN apk add --no-cache \
-  nodejs \
-  git \
-  python \
-  make \
-  g++ \
-  graphicsmagick
-RUN npm install -g databank-mongodb
-RUN npm install -g pump.io
-RUN adduser -D -g "" pumpio
+    curl \
+    nodejs \
+    python \
+    make \
+    g++ \
+    git \
+    graphicsmagick \
+ && adduser -D -g "" pumpio \
+ && mkdir -p /etc/ssl/private \
+ && npm install -g databank-mongodb@0.19.2 \
+ && mkdir -p /usr/src \
+ && curl -fSL https://github.com/pump-io/pump.io/archive/v$PUMPIO_VERSION.zip -o pumpio.zip \
```

Comment:
Yeah, I personally find the no-Docker workflow just fine but apparently a lot of people prefer to develop in Docker containers. The logic being that you set up Docker once and then don't worry about setting up a development environment again, regardless of language, dependencies, etc. See https://medium.com/iron-io-blog/why-and-how-to-use-docker-for-development-a156c1de3b24, https://docs.docker.com/opensource/project/set-up-dev-env/, etc.

However it looks like those projects don't have you rebuild the image every time... I'm not familiar with Docker enough to understand exactly how it works :(

> Also, it would make sense to decouple the docker image creation from the original source, ideally to have a separate repository for it which means that the local copy is not available anymore.

This doesn't make sense to me. I'd like to track the Docker build in the same repo so when we e.g. introduce a new dependency, we can update the Dockerfile in lockstep. Giving the Docker build a separate history makes it difficult to see that correlation. And as far as I can tell, there are no disadvantages to this approach. Am I missing something?

> Building a docker image shouldn't depend on something outside of the folder where the Dockerimage file is located (it is against the rules).

This seems to be an argument in my favor. Wouldn't your `curl` command violate that rule? :)

> A script can be made which would update the dockerfile once you publish a new version with publish, right?

Yes, but if possible I'd prefer to avoid this. Less moving parts. If it's not technically feasible though, a script can be written.